File: src/main/java/net/thedragonteam/armorplus/client/gui/GuiArmorPlusInfo.java
package net.thedragonteam.armorplus.client.gui;

import net.minecraft.client.gui.GuiScreen;
import net.thedragonteam.armorplus.APConfig;
import org.lwjgl.input.Keyboard;
import org.lwjgl.opengl.GL11;

import java.io.IOException;

import static net.thedragonteam.armorplus.util.TextUtils.formattedText;
import static net.thedragonteam.armorplus.util.Utils.setRL;

/**
 * @author Sokratis Fotkatzikis - TheDragonTeam
 */
public class GuiArmorPlusInfo extends GuiScreen {

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        int guiWidth = 256;
        int guiX = (width - guiWidth) / 2;
        int guiHeight = 256;
        int guiY = (height - guiHeight) / 2;
        GL11.glColor4f(1f, 1f, 1f, 1f);
        mc.renderEngine.bindTexture(setRL("textures/gui/gui_armorplus.png"));
        drawTexturedModalRect(guiX, guiY, 0, 0, guiWidth, guiHeight);

        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_heading"), guiX + 25, guiY + 20, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_one"), guiX + 25, guiY + 40, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_two"), guiX + 25, guiY + 50, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_three"), guiX + 25, guiY + 60, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_four"), guiX + 25, guiY + 70, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_five"), guiX + 25, guiY + 80, 0x000000);
        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_six", APConfig.gameMode), guiX + 25, guiY + 100, 0x000000);
        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    @Override
    protected void keyTyped(char typedChar, int keyCode) throws IOException {
        if (keyCode == Keyboard.KEY_E) {
            mc.displayGuiScreen(null);
        }
        super.keyTyped(typedChar, keyCode);
    }

}

File: src/main/java/net/thedragonteam/armorplus/client/gui/GuiArmorPlusInfo.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.client.gui

import net.minecraft.client.gui.GuiScreen
import net.minecraft.util.ResourceLocation
import net.thedragonteam.armorplus.ArmorPlus
import net.thedragonteam.armorplus.util.TextUtils.formattedText
import org.lwjgl.input.Keyboard
import org.lwjgl.opengl.GL11
import java.io.IOException

/**
 * net.thedragonteam.armorplus.client.gui
 * ArmorPlus created by sokratis12GR on 6/13/2516 6:42 PM.
 * - TheDragonTeam
 */
class GuiArmorPlusInfo : GuiScreen() {

    override fun drawScreen(x: Int, y: Int, ticks: Float) {
        val guiWidth = 256
        val guiX = (width - guiWidth) / 2
        val guiHeight = 256
        val guiY = (height - guiHeight) / 2
        GL11.glColor4f(1f, 1f, 1f, 1f)
        mc.renderEngine.bindTexture(ResourceLocation(ArmorPlus.MODID, "textures/gui/gui_armorplus.png"))
        drawTexturedModalRect(guiX, guiY, 0, 0, guiWidth, guiHeight)

        fontRenderer.drawString(formattedText("gui.armorplus.info.text_line_heading"), guiX + 25, guiY + 20, 0x000000)
        super.drawScreen(x, y, ticks)
    }

    @Throws(IOException::class)
    override fun keyTyped(c: Char, key: Int) {
        if (key == Keyboard.KEY_E) {
            mc.displayGuiScreen(null)
        }
        super.keyTyped(c, key)
    }

    override fun setText(newChatText: String?, shouldOverwrite: Boolean) {
        super.setText(newChatText, shouldOverwrite)
    }
}